pandera/backends/pyspark/container.py

```diff
@@ -80,6 +80,23 @@ def _column_checks(
 
         return check_obj
 
+    @validate_scope(scope=ValidationScope.DATA)
+    def _data_checks(
+        self,
+        check_obj: DataFrame,
+        schema,
+        column_info: ColumnInfo,  # pylint: disable=unused-argument
+        error_handler: ErrorHandler,
+    ):
+        """Run the checks related to data validation and uniqueness."""
+
+        # uniqueness of values
+        check_obj = self.unique(
```

Comment:
Can't rememver how this was implemented, but wouldn't it make sense to catch the error which uniqe would raise and append to the `ErrorHandler`? Similar pattern to the SchemaError above?

Reply:
Essentially catching a `DataError` or similar. @NeerajMalhotra-QB

Reply:
I agree

Reply:
Didn't get it :)
Do you agree with current implementation (this is done inside the `unique()` method) or to move it to outside of it?

Reply:
Oh, I understood the issue now, it's about the use of an `SchemaError` exception in data vadliation, not the point where it's raised. Let me check it.

Reply:
I flattened the `unique` validation to contain only one function, instead of two. The error collection is being done at root level, in the `container.py` now:
![image](https://github.com/unionai-oss/pandera/assets/110418479/bcd59574-b6bb-4a18-a303-1ae193d14540)